snuba/web/delete_query.py

```diff
@@ -27,23 +30,68 @@ def construct_condition(body: Dict[str, Any]) -> Expression:
     return combine_and_conditions(and_conditions)
 
 
+def _get_rows_to_delete(
+    storage: WritableTableStorage, select_query_to_count_rows: Query
+) -> int:
+    formatted_select_query_to_count_rows = format_query(select_query_to_count_rows)
+    select_query_results = (
+        storage.get_cluster().get_reader().execute(formatted_select_query_to_count_rows)
+    )
+    return typing.cast(int, select_query_results["data"][0]["count"])
+
+
+def _enforce_max_rows(
```

Comment:
in the architecture diagram this is a distinct component. This means that it should be tested on its own